adcircpy/server/server_config.py

```diff
@@ -0,0 +1,125 @@
+from abc import ABC
+from datetime import timedelta
+import os
+import pathlib
+from tempfile import TemporaryDirectory
+
+PADCIRC_DRIVER_SCRIPT_FILENAME = pathlib.Path(os.path.dirname(__file__)).resolve() / 'padcirc_driver.sh'
+
+argument_sbatch_translations = {
+    'run_directory': 'D',
+    'run_name'     : 'J',
+    'account'      : 'A',
+    'log_filename' : 'output',
+    'slurm_ntasks' : 'n',
+    'duration'     : 'time',
+    'partition'    : 'partition'
+}
+
+
+class ServerConfig(ABC):
+    def __init__(self, script: str):
+        self.script = script
+
+    def run(self):
+        """
+        Run the current shell script from a temporary file.
+        """
+
+        with TemporaryDirectory() as temporary_directory:
+            temporary_filename = os.path.join(temporary_directory, 'temp.job')
+            with open(temporary_filename, 'w') as temporary_file:
+                temporary_file.write(self.script)
+            os.system(temporary_filename)
+
+    def write(self, filename: str):
+        """
+        Write shell script to the given filename.
+
+        :param filename: file path to shell script
+        """
+
+        with open(filename, 'w') as output_file:
+            output_file.write(self.script)
+
+
+class SlurmScript(ServerConfig):
+    """
+    Object instance of a Slurm shell script (`*.job`).
+    """
+
+    def __init__(self, account: str, slurm_ntasks: int, run_name: str, partition: str, duration: timedelta, driver_script_filename: str = None, run_directory: str = '.', mail_type: str = None,
```

Comment:
Can we please do separate lines for long inputs? It helps with readability.

Reply:
sure thing, I can wrap this